src/Formulate.js

```diff
@@ -73,6 +75,21 @@ class Formulate {
     }
   }
 
+  /**
+   * Produce a deterministically generated id based on the sequence by which it
+   * was requested. This should be *theoretically* the same SSR as client side.
+   * However, SSR and deterministic ids can be very challenging, so this
+   * implementation is open to community review.
+   */
+  nextId (vm) {
+    const path = vm.$route && vm.$route.path && vm.$route.path || false
+    const pathPrefix = path ? vm.$route.path.replace(/[\/\\.\s]/g, '-') : 'global';
+    if (!Object.prototype.hasOwnProperty.call(this.idRegistry, pathPrefix)) {
+      this.idRegistry[pathPrefix] = 0;
+    }
+    return `${this.options.idPrefix}${pathPrefix}-${++this.idRegistry[pathPrefix]}`
```

Comment:
This is really clever. Good idea to have a global registry of IDs per path and to pass in the current VM.

Reply:
Yeah, could probably be more robust...deterministic ids are really hard. Probably the most "accurate" way would be to crawl the ancestor tree and make an id out of that, but its too expensive. Either way this is better than nanoid, and soon I think we could remove that dependency all together.